cmake/FDBComponents.cmake

```diff
@@ -18,39 +18,9 @@ endif()
 # SSL
 ################################################################################
 
-include(CheckSymbolExists)
-
-set(USE_WOLFSSL OFF CACHE BOOL "Build against WolfSSL instead of OpenSSL")
-set(USE_OPENSSL ON CACHE BOOL "Build against OpenSSL")
-if(USE_WOLFSSL)
-  set(WOLFSSL_USE_STATIC_LIBS TRUE)
-  find_package(WolfSSL)
-  if(WOLFSSL_FOUND)
-    set(CMAKE_REQUIRED_INCLUDES ${WOLFSSL_INCLUDE_DIR})
-    add_compile_options(-DHAVE_OPENSSL)
-    add_compile_options(-DHAVE_WOLFSSL)
-  else()
-    message(STATUS "WolfSSL was not found - Will compile without TLS Support")
-    message(STATUS "You can set WOLFSSL_ROOT_DIR to help cmake find it")
-    message(FATAL_ERROR "Unable to find WolfSSL")
-  endif()
-elseif(USE_OPENSSL)
-  set(OPENSSL_USE_STATIC_LIBS TRUE)
-  if(WIN32)
-    set(OPENSSL_MSVC_STATIC_RT ON)
-  endif()
-  find_package(OpenSSL)
-  if(OPENSSL_FOUND)
-    set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-    add_compile_options(-DHAVE_OPENSSL)
-  else()
-    message(STATUS "OpenSSL was not found - Will compile without TLS Support")
-    message(STATUS "You can set OPENSSL_ROOT_DIR to help cmake find it")
-    message(FATAL_ERROR "Unable to find OpenSSL")
-  endif()
-else()
-  message(FATAL_ERROR "Must set USE_WOLFSSL or USE_OPENSSL")
-endif()
+find_package(OpenSSL REQUIRED)
+set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+add_compile_options(-DHAVE_OPENSSL)
```

Comment:
Do you mean to abandon the `USE_OPENSSL` logic for windows and require openssl to be built into FDB for all users?

It also may still be helpful to tell users to set `OPENSSL_ROOT_DIR` if openssl is not found.

Reply:
`OPENSSL_ROOT_DIR` is supported by the `FindOpenSSL` module. So this should work fine.

I am in favor to require OpenSSL for FDB to build. It's just not feasible anymore for us to test all build combinations

Reply:
I believe the old logic looks like:

1. If WolfSSL, use WolfSSL
2. If OpenSSL
  a. Found: Use OpenSSL
  b. Not Found: Panic "Unable to find OpenSSL"
3. Otherwise, Panic Must set one of SSL

So it seems to me that if we drop WolfSSL, we *MUST* use OpenSSL.

Reply:
Also, in our source code, plenty of places, say "MkCert.cpp", will include <openssl/xxx.hpp> without checking any flags.

Reply:
Sounds good to me